includes/options/views/class-amp-analytics-options-serializer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+class Analytics_Options_Serializer {
+
+	public static function save() {
+
+		$option_name = 'analytics';
+
+		if ( !( empty( $_POST['vendor-type'] ) ||
+		     empty( $_POST['config'] ) ) ) {
+
+			if ( empty( $_POST['id-value'] ) ) {
+				$_POST['id-value'] = md5( $_POST['config'] );
+			}
+
+			$new_analytics_options = array(
+				$_POST['id-value'],
+				$_POST['vendor-type'],
+				stripslashes( $_POST['config'] )
```

Comment:
Done. Added a validation function and the save function proceeds only if content of config is valid JSON.